bot/exts/backend/branding/_cog.py

```diff
@@ -1,566 +1,654 @@
+
+    An event with a path of 'events/black_history_month' will resolve to 'Black History Month'.
     """
+    name = event.path.split("/")[-1]  # Inner-most directory name
+    words = name.split("_")  # Words from snake case
+
+    return " ".join(word.title() for word in words)
 
-    current_season: t.Type[_seasons.SeasonBase]
 
-    banner: t.Optional[GitHubFile]
+class Branding(commands.Cog):
+    """
+    Guild branding management.
+
+    This cog is responsible for automatic management of the guild's branding while sourcing assets directly from
+    the branding repository.
 
-    available_icons: t.List[GitHubFile]
-    remaining_icons: t.List[GitHubFile]
+    We utilize multiple Redis caches to persist state. As a result, the cog should seamlessly transition across
+    restarts without having to query either the Discord or GitHub APIs, as it will always remember which
+    assets are currently applied.
 
-    days_since_cycle: t.Iterator
+    Additionally, the state of the icon rotation is persisted. As a result, the rotation doesn't reset unless
+    the current event or its icons change.
 
-    daemon: t.Optional[asyncio.Task]
+    The cog is designed to be autonomous. The daemon, unless disabled, will poll the branding repository at
+    midnight every day and respond to detected changes. Since we persist SHA hashes of tracked assets,
+    changes in an on-going event will trigger automatic resynchronisation.
 
-    # Branding configuration
-    branding_configuration = RedisCache()
+    A #changelog notification is automatically sent when entering a new event. Changes in the branding of
+    an on-going event do not trigger a repeated notification.
+
+    The command interface allows moderators+ to control the daemon or request an asset synchronisation,
+    while regular users can see information about the current event and the overall event schedule.
+    """
+
+    # RedisCache[
+    #     "daemon_active": If True, daemon auto-starts; controlled via commands (bool)
+    #     "event_path": Path from root in the branding repo (str)
+    #     "event_description": Markdown description (str)
+    #     "event_duration": Human-readable date range or 'Fallback' (str)
+    #     "banner_hash": Hash of the last applied banner (str)
+    #     "icons_hash": Compound hash of icons in rotation (str)
+    #     "last_rotation_timestamp": POSIX timestamp (float)
+    # ]
+    cache_information = RedisCache()
+
+    # Cache holding icons in current rotation ~ the keys are download URLs (str) and the values are integers
+    # corresponding to the amount of times each icon has been used in the current rotation
+    cache_icons = RedisCache()
+
+    # Cache holding all available event names & their durations; this is cached by the daemon and read by
+    # the calendar command with the intention of preventing API spam; doesn't contain the fallback event
+    cache_events = RedisCache()
```

Comment:
We could, but I'd like to hear why. You save 18 lines at the cost of having to go to a different module to read the documentation, is that worth it?

Reply:
The current code is kind of massive, so it would be easier to view it if we break it down into multiple files. As a part of this, it would be easier to view caches if it had sit own file. One can always click on the function and go to its source (GitHub UI) or control click on Pycharm.

Reply:
For what is worth, I’d rather always have caches in the same file as the code when possible. 

I believe if the caches are in a separate file in the help channel cog for instance it is because they are used by some functions outside of the cog. I guess we could access it through the help channel but we may run into issues due to reloads and similar. Keeping them global is the easiest choice for those.

Here since the caches are used only in this cog, they should be a class attribute.

Reply:
Let's keep this in the same file.